x/ledger/client/cli/query.go

```diff
@@ -0,0 +1,468 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/version"
+
+	"github.com/provenance-io/provenance/x/ledger/helper"
+	"github.com/provenance-io/provenance/x/ledger/keeper"
+	ledger "github.com/provenance-io/provenance/x/ledger/types"
+	chunkimport "github.com/provenance-io/provenance/x/ledger/client/cli/import"
+)
+
+// GetQueryCmd is the top-level command for attribute CLI queries.
+func CmdQuery() *cobra.Command {
+	queryCmd := &cobra.Command{
+		Use:                        ledger.ModuleName,
+		Aliases:                    []string{"l"},
+		Short:                      "Querying commands for the account metadata module",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	queryCmd.AddCommand(
+		GetConfigCmd(),
+		GetLedgerEntriesCmd(),
+		GetBalancesAsOfCmd(),
+		GetLedgerClassEntryTypesCmd(),
+		GetLedgerClassStatusTypesCmd(),
+		GetLedgerClassBucketTypesCmd(),
+		GetLedgerClassCmd(),
+		GetAllSettlementsCmd(),
+		GetSettlementsByCorrelationIdCmd(),
+		chunkimport.CmdBulkImportStatus(),
+	)
+
+	return queryCmd
+}
+
+// GetAttributeParamsCmd returns the command handler for name parameter querying.
+func GetConfigCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "config <asset_class_id> <nft_id>",
```

Comment:
issue (blocking): Rename this command to `get`.

It's a tough one. The query endpoint is just "Ledger". So usually, we'd call this command `ledger`. But that's also the name of the module (and parent command). So calling it that would mean you're doing `provenanced q ledger ledger ...`. We often just call such commands "get", so that it's `provenanced q ledger get ...`.

The function name and comment should be updated to reflect the new name, too.

Reply:
I modified this to be GetCmd() and use `q ledger get` as it makes more sense than `ledger ledger` and `config` never quite worked for me.